File: Categories/Hash_Set/1_twoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var vis = new HashSet<int>();
        var rcd  = new Dictionary<int, int>();

        foreach (var (num, idx) in nums.Select((num, idx) => (num, idx))) {
            if (vis.Contains(target - num)) {
                return new int[] {idx, rcd[target - num]};
            }
            vis.Add(num);
            if (!rcd.ContainsKey(num)) rcd.Add(num, idx);
        }

        return new int[] {-1, -1};
    }
}

File: Categories/Array/1_twoSum.py
from typing import *
from collections import defaultdict

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        rcd = defaultdict(int)
        for idx, num in enumerate(nums):
            if target - num in rcd:
                return [idx, rcd[target - num]]
        
            rcd[num] = idx
        
        # error handling
        return [-1, -1]